scripts/download-browsers.js

```diff
@@ -0,0 +1,97 @@
+/*
+MIT License
+
+Copyright (c) 2021-present, Elastic NV
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+require("dotenv").config();
+
+const util = require("util");
+const path = require("path");
+const {
+  downloadBrowserWithProgressBar,
+} = require("playwright/lib/utils/browserFetcher");
+const { getChromeVersion } = require("./install-pw");
+
+const SUPPORTED_PLATFORMS = ["linux", "mac", "win"];
+const EXECUTABLE_PATHS = {
+  linux: ["chrome-linux", "chrome"],
+  mac: ["chrome-mac", "Chromium.app", "Contents", "MacOS", "Chromium"],
+  win: ["chrome-win", "chrome.exe"],
+};
+
+const CHROMIUM_DOWNLOAD_URLS = {
+  linux: "%s/builds/chromium/%s/chromium-linux.zip",
+  mac: "%s/builds/chromium/%s/chromium-mac.zip",
+  win: "%s/builds/chromium/%s/chromium-win64.zip",
+  // 'mac-arm64': '%s/builds/chromium/%s/chromium-mac-arm64.zip',
+};
+
+async function download(platform, revision, directory) {
```

Comment:
Did you try to check the Registry in Playwright if there was any other way to do this?

Reply:
Yes I went through all the Registry in Playwright and it seemed to be no work way around in there, especially I couldn't mock the platform as they would check the host machine's platform in various steps. So I used one of the functions that Registry uses to download the browser which is not tied to the host machine's platform(you pass it via parameter)